src/modules/iotjs_module_mqtt.c

```diff
@@ -74,24 +69,30 @@ static size_t get_remaining_length_size(uint32_t len) {
 }
 
 
-static uint32_t iotjs_decode_remaining_length(char *buffer, size_t *offset) {
-  unsigned char c;
-  uint32_t remaining_length = 0;
+static uint32_t iotjs_decode_remaining_length(char *buffer,
+                                              uint32_t *out_length) {
+  uint32_t decoded_length = 0;
   uint32_t length = 0;
   uint32_t shift = 0;
+  uint32_t c;
```

Comment:
Another idea: since we know that at least two bytes will be parsed, this could be reworked to:
```
uint32_t c = (uint32_t)(*buffer);
uint32_t decoded_length = (c & 0x7F);
uint32_t shift = 7;
uint32_t length = 1;

buffer++;
```

Reply:
Great idea!